solvebio/resource/object.py

```diff
@@ -284,9 +285,18 @@ def _archive(self, archive_folder):
         # Create timestamped archive filename
         date_format = "%Y-%m-%d_%Hh%Mm%Ss_%Z"
         timestamp = datetime.now().strftime(date_format)
-        file_extension = self.filename.split(".")[-1]
-        archive_filename = ".".join(self.filename.split(".")[:-1]) \
-                + "_" + timestamp + "." + file_extension
+        base_filename, file_extension = os.path.splitext(self.filename)
```

Comment:
Does it make sense to move some of this into utils/files.py 🤣

Reply:
For sure lol.. done!